blog/_posts/2019-02-17-github-situation.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title:  "The Github situation"
+date:   2019-02-17
+author: nezorflame
+---
+<div markdown='1'>
+![Github]({{ site.blogimg }}github.png)
+</div>
+Hello everyone!
+My name is Ilya Danilkin aka [@nezorflame]. You may know me from my work on the unofficial Oreo/Pie OpenGApps builds at XDA and the support group in Telegram.
+
+On the behalf of the OpenGApps team I'd like to explain the recent problems with our website being unable to serve downloads and the takedown of our application repos at Github.
+
+### The project structure
+
+To understand the problem, first we should take a look at how the OpenGApps are structured and built.
+We have multiple repos hosted on Github at <https://github.com/opengapps> which are being used by the buildserver to automatically produce the builds.
+
+The current structure consists of two parts: script repository and the APK repositories.
+
+#### Script repository ([/opengapps])
+
+This is the main repo with all the scripts needed to compile an OpenGApps flashable package.
+At the [/sources](https://github.com/opengapps/opengapps/tree/master/sources) folder we have 5 subrepos which are being pulled before the build, thus downloading the necessary APKs for the preferred architecture (let's call those subrepos as **APK repos**).
+E.g. to build an ARM64 package, we'd need [/all], [/arm] and [/arm64] repos as some of the apps don't have 64-bit version yet.
+
+#### APK repositories ([/all], [/arm], [/arm64], [/x86] and [/x86_64])
+
+These are the repositories which are holding all the required APKs. They are being automatically extracted from the Google Pixel [factory images](https://developers.google.com/android/ota) every time they get updated.
+
+These repos are huge in size:
+
+- **all**: 4.99 GB
+- **arm**: 147.47 GB
+- **arm64**: 93.81 GB
+- **x86**: 67.17 GB
+- **x86_64**: 26.61 GB
+
+### The problem
+
+Last week (starting February 10th) Github has closed all our APK repos on the account of breaking the [ToS](https://help.github.com/articles/github-terms-of-service/). Namely, the paragraph *C. Acceptable Use* at *4. Services Usage Limits* part. Turns out, they weren't happy with us storing all these big APK files inside of the repositories.
```

Comment:
Please add something about being thankful for GitHub for hosting us so long, even when the repository was already big, and looking constructively for possible solutions

Reply:
Added